### ISSUE TITLE
--force-formatter is passed to a version of chef-client that does not support it.

### ISSUE THREAD

**Issue by damm** [test-kitchen/test-kitchen#593]
```
/usr/local/lib64/ruby/gems/1.9.1/gems/mixlib-cli-1.5.0/lib/mixlib/cli.rb:191:in `parse_options': invalid option: --force-formatter (OptionParser::InvalidOption)

       	from /usr/local/lib64/ruby/gems/1.9.1/gems/chef-10.16.4/lib/chef/application.rb:64:in `reconfigure'
       	from /usr/local/lib64/ruby/gems/1.9.1/gems/chef-10.16.4/lib/chef/application/solo.rb:155:in `reconfigure'
       	from /usr/local/lib64/ruby/gems/1.9.1/gems/chef-10.16.4/lib/chef/application.rb:70:in `run'
       from /usr/local/lib64/ruby/gems/1.9.1/gems/chef-10.16.4/bin/chef-solo:25:in `<top (required)>'
       	from /opt/chef/bin/chef-solo:23:in `load'
       	from /opt/chef/bin/chef-solo:23:in `<main>'
```  

See https://github.com/test-kitchen/test-kitchen/blob/a83398eba6d850971805447a42270925847acc08/lib/kitchen/provisioner/chef_solo.rb#L50

While I don't disagree with the intent; I am in a special edge case where I'm trying to support an older version of Chef (that should have been upgraded years ago).

**Comment by BobbyRyterski** [comment]
I'd also like to see this fixed.

**Comment by edmundcraske** [comment]
Indeed, should be easy enough to keep backward compatibility here.

**Comment by lamont-granquist** [comment]
Chef 10 is EOL and no longer supported.

**Comment by fnichol** [comment]
If someone on this thread can get the version of Chef where this flag is introduced, I think this can be a simple enough PR. Any helpers?

**Comment by lamont-granquist** [comment]
Should be 11.0.0

**Comment by BobbyRyterski** [comment]
Looks like `--force-formatter` was added in [this commit](https://github.com/chef/chef/commit/1c0e6c22ebf025de71c8d8fde8c273bd654f1553), first available in 11.0.0.

**Comment by fnichol** [comment]
Excellent, thanks all!

**Comment by Ajedi32** [comment]
So what's the workaround for this? Just downgrade to an older version of Chef on the host machine?

**Comment by lamont-granquist** [comment]
upgrade Chef on the virt you're testing.  10.x is EOL and no longer supported.

but should probably be closed as WONTFIX unless someone wants to submit a PR to fix it.

**Comment by cheeseplus** [comment]
Closing this as WONTFIX - Chef 10 is EOL but if someone wants to make a PR I'd not mind reviewing.

**Comment by adamdecaf** [comment]
FYI This was fixed in vagrant 1.8.0 https://github.com/mitchellh/vagrant/blob/master/CHANGELOG.md#180-december-21-2015

**Comment by Lumbendil** [comment]
@adamdecaf seems like it stopped working on 1.8.1

**Comment by acondrat** [comment]
I submitted https://github.com/test-kitchen/test-kitchen/pull/1072 to enable Chef 10 support.

**Comment by Ajedi32** [comment]
Similar issues/PRs in Vagrant (for future reference): mitchellh/vagrant#6278 mitchellh/vagrant#6556 (Fix released in v1.8.0.)
